lms/services/blackboard_api/service.py

```diff
@@ -53,12 +53,38 @@ def list_files(self, course_id):
         :raise ProxyAPIAccessTokenError: if we don't have a Blackboard API
             access token for the current user
         """
-        return self._http_service.get(
+        files = self._http_service.get(
             self._api_url(f"courses/uuid:{course_id}/resources"),
             oauth=True,
             schema=BlackboardListFilesSchema,
         ).validated_data
 
+        return [
+            {
+                "id": self._blackboard_document_url(file_["id"]),
```

Comment:
This is where we turn Blackboard file IDs into `blackboard://...` URLs before returning the list of them to the frontend when a teacher is creating an assignment.

They are just opaque strings to the frontend, it knows nothing about the `blackboard://...` format.

When the teacher selects a file the frontend submits the `blackboard://...` URL to the `configure_module_item()` view as the `document_url` and it gets saved in the DB. It's an opaque string to the `configure_module_item()` view and to the DB as well.

One weird thing about this is that `blackboard://...` is the file `id` in the list files response from our proxy API but it's the `document_url` in the `ModuleItemConfiguration` table :man_shrugging: 

Note that we only every format `blackboard://...` URLs. We never have to parse them :)

Also this `_blackboard_document_url()` methodis almost the only place in the code that knows about the `blackboard://...` format. The only exception to (2) is that `JSConfig` has to look for strings that start with `blackboard://` to know when a Blackboard Files assignment is being launched. These are the only two places in the code where the `"blackboard://"` string exists.

https://github.com/hypothesis/lms/blob/1a0324d8e0e8213bb0f27c46a2561e20206a17af/lms/resources/_js_config.py#L65-L73